### DISCUSSION TITLE
Error computing velocity embedding when providing own sequences

### DISCUSSION THREAD

**Discussion by DrewBarratt** [brianhie/evolocity#22] [Q&A]
Hi, 
After several attempts with different sequences, I am facing the same error repeatedly. Any advice on this would be greatly appreciated. 
 
Error:

computing velocity embedding
    finished (0:00:00) --> added
    'velocity_umap', embedded velocity vectors (adata.obsm)
Traceback (most recent call last):
  File "test.py", line 17, in <module>
    evo.pl.velocity_embedding_grid(adata)
  File "/miniconda/venv/lib/python3.7/site-packages/evolocity/plotting/velocity_embedding_grid.py", line 206, in velocity_embedding_grid
    min_mass=min_mass,
  File "/miniconda/venv/lib/python3.7/site-packages/evolocity/plotting/velocity_embedding_grid.py", line 51, in compute_velocity_on_grid
    nn.fit(X_emb)
  File "/miniconda/venv/lib/python3.7/site-packages/sklearn/neighbors/_unsupervised.py", line 166, in fit
    return self._fit(X)
  File "/miniconda/venv/lib/python3.7/site-packages/sklearn/neighbors/_base.py", line 569, in _fit
    raise ValueError("Expected n_neighbors > 0. Got %d" % self.n_neighbors)
ValueError: Expected n_neighbors > 0. Got 0

with the following python script. 

import evolocity as evo
import scanpy as sc
fasta_fname = 'data.fasta'
adata = evo.pp.featurize_fasta(fasta_fname)
evo.pp.neighbors(adata)
evo.tl.velocity_graph(adata)
sc.tl.umap(adata)
evo.tl.velocity_embedding(adata)
evo.pl.velocity_embedding_grid(adata)
evo.pl.velocity_embedding_stream(adata)

**Comment by brianhie** [comment]
Can you verify that the tutorial works? E.g., https://evolocity.readthedocs.io/en/latest/evolocity_nucleoprotein.html

**Comment by DrewBarratt** [comment]
Hi Brian, 

Thanks for getting back to me. 

Yes, I was able to run the example files however I now believe the issues has now been resolved. 

We reduced the n_neighbors value=28 which seems to have solved this error. I think it is most likely a result of the low variability within my dataset. Out of interest, do you know if there is an optimal n_neighbors value? In the code you suggest that it can be set to anything 2-100 but do we know the effects of choosing either extreme value?

Thanks again/ Diolch

**Comment by brianhie** [comment]
Generally, higher nearest neighbors values helps to smooth out the landscape. 28 should be fine -- it's also generally good practice to see that any conclusions about the landscape are reasonably robust to different values of that parameter.

**Comment by haoyuewang27** [comment]
Hi, I am not an expert in coding and am trying to use eco-velocity on my protein analysis properly. I did everything following the tutorial, but did not find where to change the n_neighbors. Can you tell me where to change n_neighbors values? Thank you!

**Reply by haoyuewang27** [reply]
NVM I found it in the comments of  neighbors.py !
